From ef21e9c7936f6dfdb00d7e47842947e4423c36a4 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Date: Thu, 18 Jun 2020 10:02:52 -0700
Subject: [PATCH 1180/1239] qlm command: Fix prbs command syntax

Allow all parameters to be passed, run prbs only on the lane
provided.

Change-Id: Ia0e28b36488b878154e9fce31327179e4fc0e453
Signed-off-by: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/boot/u-boot/+/30326
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
---
 cmd/marvell/octeontx2_qlm_cmds.c | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/cmd/marvell/octeontx2_qlm_cmds.c b/cmd/marvell/octeontx2_qlm_cmds.c
index 60dbbdcf5b..caa40abcbf 100644
--- a/cmd/marvell/octeontx2_qlm_cmds.c
+++ b/cmd/marvell/octeontx2_qlm_cmds.c
@@ -21,14 +21,9 @@ static int do_prbs(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 
 	if (argc == 5) {
 		qlm = simple_strtoul(argv[1], NULL, 10);
-		mode = simple_strtoul(argv[2], NULL, 10);
-		time = simple_strtoul(argv[3], NULL, 10);
-		lane = simple_strtoul(argv[4], NULL, 10);
-	} else if (argc == 4) {
-		qlm = simple_strtoul(argv[1], NULL, 10);
-		mode = simple_strtoul(argv[2], NULL, 10);
-		time = simple_strtoul(argv[3], NULL, 10);
-		lane = -1;
+		lane = simple_strtoul(argv[2], NULL, 10);
+		mode = simple_strtoul(argv[3], NULL, 10);
+		time = simple_strtoul(argv[4], NULL, 10);
 	} else {
 		return CMD_RET_USAGE;
 	}
@@ -40,10 +35,10 @@ static int do_prbs(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 
 U_BOOT_CMD(prbs,    5,    1,     do_prbs,
 	   "command to run PRBS on slected QLM",
-	   "<qlm> <mode> <time> [lane]\n"
+	   "<qlm> <lane> <mode> <time> \n"
 	   "    - run PRBS with pattern indicated by 'mode' on selected 'qlm'\n"
 	   "      PRBS will be enabled by 'time' seconds\n"
-	   "      PRBS is performed on a particular LANE if passed"
+	   "      PRBS is performed on a particular 'lane'"
 );
 
 static int do_eye(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
-- 
2.29.0

